extensions/wikia/Search/classes/Test/Controller/SearchControllerTest.php

```diff
@@ -409,159 +409,32 @@ public function testHandleArticleMatchTrackingWithoutGoSearch() {
-		$mockTitle			=	$this->getMockBuilder( 'Title' )->disableOriginalConstructor()->getMock();
-		$mockResponse		=	$this->getMock( 'WikiaResponse', array( 'redirect', 'setVal' ), array( 'html' ) );
-		$mockRequest		=	$this->getMock( 'WikiaRequest', array( 'getVal' ), array( array() ) );
-		$configMethods		=	array( 'getResultsFound', 'getPage', 'getQuery', 'getNumPages', 'getInterWiki',
-										'getSkipCache', 'getDebug', 'getNamespaces', 'getAdvanced', 'getLimit' );
-		$mockConfig			=	$this->getMock( 'Wikia\Search\Config', $configMethods );
-
-		$mockWgRefl = new ReflectionProperty( 'WikiaSearchController', 'wg' );
-		$mockWgRefl->setAccessible( true );
-		$mockWgRefl->setValue( $mockController, (object) array( 'Title' => $mockTitle ) );
-
-		$mockController
-			->expects	( $this->any() )
-			->method	( 'getVal' )
-			->with		( 'config', false )
-			->will		( $this->returnValue( false ) )
-		;
-		$e = null;
-		try {
-			$mockController->pagination();
-			$this->assertFalse(
-					true,
-					'WikiaSearchController::pagination should throw an exception if the "config" is not set in the request.'
-			);
-		} catch ( Exception $e ) { }
-
-		$this->assertInstanceOf(
-				'Exception',
-				$e,
-				'WikiaSearchController::pagination should throw an exception if an instance of Wikia\Search\Config is not set in the request'
-		);
+		$intRes = $this->app->sendRequest( WikiaSearchController::class, 'pagination' );
+		$this->assertEmpty( $intRes->getBody(), 'Search controller pagination must render no response if no config was supplied' );
 	}
 
 	/**
-	 * @group Slow
-	 * @slowExecutionTime 0.07875 ms
 	 * @covers WikiaSearchController::pagination
 	 */
-	public function testPaginationMalformedConfig() {
-
-		$mockController		=	$this->searchController->setMethods( array( 'getVal', 'setVal' ) )->getMock();
-		$mockTitle			=	$this->getMockBuilder( 'Title' )->disableOriginalConstructor()->getMock();
-		$mockResponse		=	$this->getMock( 'WikiaResponse', array( 'redirect', 'setVal' ), array( 'html' ) );
-		$mockRequest		=	$this->getMock( 'WikiaRequest', array( 'getVal' ), array( array() ) );
-		$configMethods		=	array( 'getResultsFound', 'getPage', 'getQuery', 'getNumPages', 'getInterWiki',
-										'getSkipCache', 'getDebug', 'getNamespaces', 'getAdvanced', 'getLimit' );
-		$mockConfig			=	$this->getMock( 'Wikia\Search\Config', $configMethods );
+	public function testPaginationWithConfigNoResults() {
+		$configMock = $this->getMock( Wikia\Search\Config::class, [ 'getResultsFound' ] );
+		$configMock->expects( $this->once() )
+			->method( 'getResultsFound' )
+			->willReturn( 0 );
 
-		$mockWgRefl = new ReflectionProperty( 'WikiaSearchController', 'wg' );
-		$mockWgRefl->setAccessible( true );
-		$mockWgRefl->setValue( $mockController, (object) array( 'Title' => $mockTitle ) );
+		$res = $this->app->sendRequest( WikiaSearchController::class, 'pagination', [
+			'config' => $configMock
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "config" 9 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)